tests/test_slurm_command_gen_strategy.py

```diff
@@ -185,6 +185,47 @@ def test_generate_full_srun_command(self, strategy_fixture: SlurmCommandGenStrat
         full_srun_command = strategy_fixture.generate_full_srun_command({}, {}, {}, "")
         assert full_srun_command == " \\\n".join(["srun", "--test", "test_arg", "test_command"])
 
+    def test_generate_full_srun_command_with_run_pre_test_flag(
+        self, jax_strategy_fixture: JaxToolboxSlurmCommandGenStrategy, slurm_system: SlurmSystem
+    ):
+        # Assign the slurm_system to the strategy fixture
+        jax_strategy_fixture.slurm_system = slurm_system
+
+        # Mock necessary methods and attributes
+        jax_strategy_fixture._create_run_script = MagicMock()
+        jax_strategy_fixture._generate_pre_test_command = MagicMock(return_value="pre_test_command")
+        jax_strategy_fixture._generate_pre_test_check_command = MagicMock(return_value="pre_test_check_command")
+
+        slurm_args = {
+            "output": "output.txt",
+            "error": "error.txt",
+            "image_path": "image_path",
+            "container_mounts": "container_mounts",
+        }
+        env_vars = {}
+        extra_cmd_args = ""
+
+        # run_pre_test is True
+        cmd_args = {"output_path": "/path/to/output", "pre_test": "true"}
+        result = jax_strategy_fixture.generate_full_srun_command(slurm_args, env_vars, cmd_args, extra_cmd_args)
+        assert "pre_test_command" in result
+        assert "pre_test_check_command" in result
+        assert "--mpi=fake-mpi" in result
```

Comment:
IMO we can make this test more reliable with `assert f"--mpi={slurm_system.mpi}" in result` and similar for the rest two asserts.

Reply:
done